docs/monitors/jmx.md

```diff
@@ -115,6 +115,37 @@ output.sendDatapoints([
 
 ```
 
+This script collects the same JMX metrics as emitted by default from the [genericjmx monitor](collectd-genericjmx.md).
+It can be used as a starting point for collecting additional JMX metrics.
+
+```groovy
+Map dims(Map args) {
+    // so that dashboard will import
+    d = ["plugin": "GenericJMX"]
+    d.putAll(args ?: [:])
+    d
+}
+dps = [
+    util.makeGauge("gauge.jvm.threads.count", util.queryJMX("java.lang:type=Threading").first().ThreadCount, dims()),
+    util.makeGauge("gauge.loaded_classes", util.queryJMX("java.lang:type=ClassLoading").first().LoadedClassCount, dims()),
+]
+mem = util.queryJMX("java.lang:type=Memory").first()
+memoryPool = util.queryJMX("java.lang:type=MemoryPool,*")
+for (i in ["committed", "init", "max", "used"]) {
+    metric = "jmx_memory.$i"
+    dps << util.makeGauge(metric, mem.HeapMemoryUsage."$i", dims(plugin_instance: "memory-heap"))
```

Comment:
Please no more `plugin_instance`.  This is one of the worst things about collectd metrics in that we have no idea what the "plugin_instance" is actually referring to.  Use `pool_name` or something a million times more descriptive.